### ISSUE TITLE
cmake invalid new-expression of abstract class type 

### ISSUE THREAD

**Issue by cnut1648** [dmlc/decord#59]
Hi, when installing from the source (I want to use with cuda) and running `cmake` there is the following error
```
Scanning dependencies of target decord
[  2%] Building CXX object CMakeFiles/decord.dir/src/runtime/c_runtime_api.cc.o
[  5%] Building CXX object CMakeFiles/decord.dir/src/runtime/cpu_device_api.cc.o
[  8%] Building CXX object CMakeFiles/decord.dir/src/runtime/dso_module.cc.o
[ 11%] Building CXX object CMakeFiles/decord.dir/src/runtime/file_util.cc.o
[ 14%] Building CXX object CMakeFiles/decord.dir/src/runtime/module.cc.o
[ 17%] Building CXX object CMakeFiles/decord.dir/src/runtime/module_util.cc.o
[ 20%] Building CXX object CMakeFiles/decord.dir/src/runtime/ndarray.cc.o
[ 22%] Building CXX object CMakeFiles/decord.dir/src/runtime/registry.cc.o
[ 25%] Building CXX object CMakeFiles/decord.dir/src/runtime/str_util.cc.o
[ 28%] Building CXX object CMakeFiles/decord.dir/src/runtime/system_lib_module.cc.o
[ 31%] Building CXX object CMakeFiles/decord.dir/src/runtime/thread_pool.cc.o
[ 34%] Building CXX object CMakeFiles/decord.dir/src/runtime/threading_backend.cc.o
[ 37%] Building CXX object CMakeFiles/decord.dir/src/runtime/workspace_pool.cc.o
[ 40%] Building CXX object CMakeFiles/decord.dir/src/sampler/random_file_order_sampler.cc.o
[ 42%] Building CXX object CMakeFiles/decord.dir/src/sampler/random_sampler.cc.o
[ 45%] Building CXX object CMakeFiles/decord.dir/src/sampler/sequential_sampler.cc.o
[ 48%] Building CXX object CMakeFiles/decord.dir/src/sampler/smart_random_sampler.cc.o
[ 51%] Building CXX object CMakeFiles/decord.dir/src/video/logging.cc.o
[ 54%] Building CXX object CMakeFiles/decord.dir/src/video/storage_pool.cc.o
[ 57%] Building CXX object CMakeFiles/decord.dir/src/video/video_interface.cc.o
[ 60%] Building CXX object CMakeFiles/decord.dir/src/video/video_loader.cc.o
[ 62%] Building CXX object CMakeFiles/decord.dir/src/video/video_reader.cc.o
/home/chris/data/untitled/python/data/pytorch/decord/src/video/video_reader.cc: In member function ‘virtual void decord::VideoReader::SetVideoStream(int)’:
/home/chris/data/untitled/python/data/pytorch/decord/src/video/video_reader.cc:101:62: error: invalid new-expression of abstract class type ‘decord::cuda::CUThreadedDecoder’
  101 |             ctx_.device_id, codecpar.get(), fmt_ctx_->iformat));
      |                                                              ^
In file included from /home/chris/data/untitled/python/data/pytorch/decord/src/video/video_reader.cc:10:
/home/chris/data/untitled/python/data/pytorch/decord/src/video/nvcodec/cuda_threaded_decoder.h:29:7: note:   because the following virtual functions are pure within ‘decord::cuda::CUThreadedDecoder’:
   29 | class CUThreadedDecoder final : public ThreadedDecoderInterface {
      |       ^~~~~~~~~~~~~~~~~
In file included from /home/chris/data/untitled/python/data/pytorch/decord/src/video/video_reader.h:10,
                 from /home/chris/data/untitled/python/data/pytorch/decord/src/video/video_reader.cc:7:
/home/chris/data/untitled/python/data/pytorch/decord/src/video/threaded_decoder_interface.h:21:22: note: 	‘virtual void decord::ThreadedDecoderInterface::SetCodecContext(AVCodecContext*, int, int, int)’
   21 |         virtual void SetCodecContext(AVCodecContext *dec_ctx, int width = -1, int height = -1, int rotation = 0) = 0;
      |                      ^~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/decord.dir/build.make:356: CMakeFiles/decord.dir/src/video/video_reader.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:96: CMakeFiles/decord.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
```
My machine is manjaro with cuda 10.2 and cmake version 3.17.2.

Any advice?

**Comment by sbebo** [comment]
Should be fixed by #58

**Comment by zhreshold** [comment]
Closed by #58
